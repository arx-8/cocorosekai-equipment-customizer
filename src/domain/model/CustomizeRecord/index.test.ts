import {
  EquipCost,
  Equipment,
  EquipmentId,
  HitPointNum,
  MagicAttackNum,
  MagicDefenseNum,
  PhysicalAttackNum,
  PhysicalDefenseNum,
} from "src/domain/model/Equipment"

import { calcCustomizeRecord } from "."

describe("calcCustomizeRecord", () => {
  it("list length 0", () => {
    // ## Arrange ##
    const equipments: Equipment[] = []

    // ## Act ##
    const result = calcCustomizeRecord(equipments)

    // ## Assert ##
    expect(result).toMatchSnapshot()
  })

  it("list length 1", () => {
    // ## Arrange ##
    const equipments: Equipment[] = [
      {
        attribute: "light",
        enhancementsNum: 0,
        equipCost: 12 as EquipCost,
        id: 0 as EquipmentId,
        imageUrl: "https://example.com/",
        name: "羊飼いの杖",
        rawName: "羊飼いの杖",
        specialEffects: [
          {
            rawText: "受ける魔法ダメージ5%軽減",
          },
        ],
        statuses: {
          hp: 72 as HitPointNum,
          magicAtk: 43 as MagicAttackNum,
          magicDef: 0 as MagicDefenseNum,
          physicalAtk: 14 as PhysicalAttackNum,
          physicalDef: 0 as PhysicalDefenseNum,
        },
      },
    ]

    // ## Act ##
    const result = calcCustomizeRecord(equipments)

    // ## Assert ##
    expect(result).toMatchSnapshot()
  })

  it("list some length", () => {
    // ## Arrange ##
    const equipments: Equipment[] = [
      {
        attribute: "darkness",
        enhancementsNum: 1,
        equipCost: 100_000 as EquipCost,
        id: 1 as EquipmentId,
        imageUrl: null,
        name: "dummy1",
        rawName: "dummy+",
        specialEffects: [
          {
            rawText: "effect 1",
          },
        ],
        statuses: {
          hp: 1 as HitPointNum,
          magicAtk: 10 as MagicAttackNum,
          magicDef: 100 as MagicDefenseNum,
          physicalAtk: 1_000 as PhysicalAttackNum,
          physicalDef: 10_000 as PhysicalDefenseNum,
        },
      },
      {
        attribute: "light",
        enhancementsNum: 2,
        equipCost: 2_000_000 as EquipCost,
        id: 20 as EquipmentId,
        imageUrl: null,
        name: "dummy2",
        rawName: "dummy++",
        specialEffects: [],
        statuses: {
          hp: 20 as HitPointNum,
          magicAtk: 200 as MagicAttackNum,
          magicDef: 2_000 as MagicDefenseNum,
          physicalAtk: 20_000 as PhysicalAttackNum,
          physicalDef: 200_000 as PhysicalDefenseNum,
        },
      },
      {
        attribute: "darkness",
        enhancementsNum: 2,
        equipCost: 30_000_000 as EquipCost,
        id: 300 as EquipmentId,
        imageUrl: "https://example.com/",
        name: "dummy3",
        rawName: "dummy++",
        specialEffects: [
          {
            rawText: "effect 300",
          },
        ],
        statuses: {
          hp: 300 as HitPointNum,
          magicAtk: 3_000 as MagicAttackNum,
          magicDef: 30_000 as MagicDefenseNum,
          physicalAtk: 300_000 as PhysicalAttackNum,
          physicalDef: 3_000_000 as PhysicalDefenseNum,
        },
      },
    ]

    // ## Act ##
    const result = calcCustomizeRecord(equipments)

    // ## Assert ##
    expect(result).toMatchSnapshot()
  })
})
