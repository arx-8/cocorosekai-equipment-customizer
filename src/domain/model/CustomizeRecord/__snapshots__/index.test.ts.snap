// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calcCustomizeRecord list length 0 1`] = `
Object {
  "equippedIds": Array [],
  "mixedAttributes": Array [],
  "mixedSpecialEffects": Array [],
  "totalEquipCost": 0,
  "totalSpecValue": 0,
  "totalStatuses": Object {
    "hp": 0,
    "magicAtk": 0,
    "magicDef": 0,
    "physicalAtk": 0,
    "physicalDef": 0,
  },
}
`;

exports[`calcCustomizeRecord list length 1 1`] = `
Object {
  "equippedIds": Array [
    0,
  ],
  "mixedAttributes": Array [
    "light",
  ],
  "mixedSpecialEffects": Array [
    Object {
      "rawText": "受ける魔法ダメージ5%軽減",
    },
  ],
  "totalEquipCost": 12,
  "totalSpecValue": 0,
  "totalStatuses": Object {
    "hp": 72,
    "magicAtk": 43,
    "magicDef": 0,
    "physicalAtk": 14,
    "physicalDef": 0,
  },
}
`;

exports[`calcCustomizeRecord list some length 1`] = `
Object {
  "equippedIds": Array [
    1,
    20,
    300,
  ],
  "mixedAttributes": Array [
    "darkness",
    "light",
  ],
  "mixedSpecialEffects": Array [
    Object {
      "rawText": "effect 1",
    },
    Object {
      "rawText": "effect 300",
    },
  ],
  "totalEquipCost": 32100000,
  "totalSpecValue": 0,
  "totalStatuses": Object {
    "hp": 321,
    "magicAtk": 3210,
    "magicDef": 32100,
    "physicalAtk": 321000,
    "physicalDef": 3210000,
  },
}
`;
